nixos/modules/virtualisation/nixos-containers.nix

```diff
@@ -518,7 +518,7 @@ in
 
     boot.enableContainers = mkOption {
       type = types.bool;
-      default = true;
+      default = config.containers != { };
```

Comment:
When the default evaluates to something using `config`, you need to include a `defaultText`.

also the description needs an update, since it no longer defaults to `true` and it's not zero cost if it's set to `true` anyway.